src/SourceGenerators/Uno.UI.SourceGenerators/ImplementedRoutedEvents/ImplementedRoutedEventsGenerator.cs

```diff
@@ -0,0 +1,251 @@
+		{
+			if (typeDeclaration.Modifiers.Any(SyntaxKind.PartialKeyword))
+			{
+				// We have partial modifier, now we check the parent.
+				// If the parent isn't TypeDeclarationSyntax (e.g, CompilationUnitSyntax or NamespaceDeclarationSyntax), then
+				// we are good. Otherwise, recurse.
+				return typeDeclaration.Parent is not TypeDeclarationSyntax parent || IsPartialChainOfTypes(parent);
+			}
+
+			return false;
+		}
+
+		private void ProcessType(INamedTypeSymbol type)
+		{
+			if (!type.IsAbstract &&
+				type.Is(_uiElementSymbol) &&
+				type.DeclaringSyntaxReferences[0].GetSyntax() is TypeDeclarationSyntax typeDeclaration &&
+				IsPartialChainOfTypes(typeDeclaration))
+			{
+				var builder = new StringBuilder("global::Uno.UI.Xaml.RoutedEventFlag.None");
+				
+				foreach (var @event in _events)
+				{
+					if (!@event.Key.StartsWith("On", StringComparison.Ordinal))
+					{
+						throw new InvalidOperationException("Expected event to start with 'On'");
+					}
+
+					if (IsEventOverrideImplemented(type, @event.Key, @event.Value))
+					{
+						builder.Append($" | global::Uno.UI.Xaml.RoutedEventFlag.{@event.Key.Substring("On".Length)}");
+					}
+				}
+
+				GenerateCode(type, builder.ToString());
+			}
+		}
+
+		private bool IsEventOverrideImplemented(INamedTypeSymbol type, string name, INamedTypeSymbol arg)
+		{
+			if (type.Equals(_uiElementSymbol, SymbolEqualityComparer.Default) ||
+				type.Equals(_controlSymbol, SymbolEqualityComparer.Default))
+			{
+				return false;
+			}
+
+			var method = type
+				.GetMembers(name)
+				.OfType<IMethodSymbol>()
+				.SingleOrDefault(
+					method =>
+						method.Parameters.Length == 1 &&
+						arg.Equals(method.Parameters[0].Type, SymbolEqualityComparer.Default));
+
+			// Base type can't be null, so the suppression should be safe.
+			// We know that the initial type inherits UIElement class (ie, _implementedRoutedEvents.ContainingType)
+			// And the recursion will end as soon as we get there.
+			return
+				method?.IsOverride == true ||
+				IsEventOverrideImplemented(type.BaseType!, name, arg);
+		}
+
+		private void GenerateCode(INamedTypeSymbol type, string routedEventFlags)
+		{
+			var builder = new IndentedStringBuilder();
+			builder.AppendLineIndented("// <auto-generated>");
+
+			var stack = type.AddToIndentedStringBuilder(builder);
+
+			WriteClass(builder, type, routedEventFlags);
+
+			while (stack.Count > 0)
+			{
+				stack.Pop().Dispose();
+			}
+
+			// Keep containing namespace here to avoid controls defined in both WUXC and MUXC from one overwriting the other.
+			var generatedSource = builder.ToString();
+			var generatedFileName = $"{GetFullMetadataNameForFileName(type)}_ImplementedRoutedEvents.g.cs";
+			_context.AddSource(generatedFileName, generatedSource);
+		}
+
+		private static void WriteClass(IIndentedStringBuilder builder, INamedTypeSymbol type, string routedEventFlags)
+		{
+			// TODO MZ: Handle generics
```

Comment:
@MartinZikmund Is `Given_Generic_Control_In_Global_Namespace` test enough to allow us to delete this? Or is there something else missing?

Reply:
I guess it should be 🤔